eras/conway/impl/src/Cardano/Ledger/Conway/Governance/DRepPulser.hs

```diff
@@ -173,25 +174,42 @@ instance EraPParams era => FromCBOR (PulsingSnapshot era) where
 --   (b) is the size of the StakeDistr, and
 --   (c) is the size of the DRepDistr, this grows as the accumulator
 computeDRepDistr ::
-  forall c.
-  Map (Credential 'Staking c) (CompactForm Coin) ->
-  Map (Credential 'DRepRole c) (DRepState c) ->
-  Map (DRep c) (CompactForm Coin) ->
-  Map (Credential 'Staking c) (UMElem c) ->
-  Map (DRep c) (CompactForm Coin)
-computeDRepDistr stakeDistr regDReps dRepDistr uMapChunk =
+  forall era.
+  Map (Credential 'Staking (EraCrypto era)) (CompactForm Coin) ->
+  Map (Credential 'DRepRole (EraCrypto era)) (DRepState (EraCrypto era)) ->
+  StrictSeq (GovActionState era) ->
+  Map (DRep (EraCrypto era)) (CompactForm Coin) ->
+  Map (Credential 'Staking (EraCrypto era)) (UMElem (EraCrypto era)) ->
+  Map (DRep (EraCrypto era)) (CompactForm Coin)
+computeDRepDistr stakeDistr regDReps gass dRepDistr uMapChunk =
   Map.foldlWithKey' go dRepDistr uMapChunk
   where
+    -- QUESTION: Can I memoize this as a field in DRepPulsar? That causes a
+    -- type-error, though that does _not_ get resolved even with AllowAmbiguousTypes
+    -- as `EraCrypto era` is a non-injective type-family
```

Comment:
So you mean, in DRepPulser, instead of having
```dpProposals :: !(StrictSeq (GovActionState era))`, to have a map from credential to deposit?

Reply:
The answer is it is an absolutele must to compute this during the DRep pulser setup in: `setFreshDRepPulsingState`. Feel free to ping me with a type error if you can't figure it out. 

You might need a `forall` in the `setFreshDRepPulsingState` and a type annotation on the field, but `AllowAmbiguousTypes` is definitely not the right solution here

Reply:
This is now implemented, but I had to use `Proxy` :+1: